core/src/message/init.rs

```diff
@@ -131,9 +131,20 @@ pub struct InitPacket {
 
 impl InitPacket {
     /// Create new InitPacket without gas.
-    pub fn new(code_id: CodeId, salt: Salt, payload: Payload, value: Value) -> Self {
+    pub fn new(
+        code_id: CodeId,
+        salt: Salt,
+        payload: Payload,
+        message_id: Option<MessageId>,
```

Comment:
please split it and `new_with_gas` into two different functions as it did for ProgramId generation

Reply:
Cannot, this will cause the `host` macro to break if I try to change these definitions.

Reply:
As we discussed, don't use host macro for the case

Reply:
Should I remove the #[host] from the init packet function in funcs.rs and paste in what it would expand into? This would be inconsistent with all the other functions in funcs.rs. I think removing the host macro should be its own issue, or changing it such that it works with this, since that is also possible.

Reply:
For me: Temporary copy-paste > bad design and usage of the structs
@SabaunT

Reply:
`host` is to be dead, so let's postpone renaming until we remove `host`.

btw, @mqxf you can remove `host` in a separate PR.